OsmAnd/src/net/osmand/plus/search/NearbyPlacesAdapter.kt

```diff
@@ -107,22 +109,36 @@ class NearbyPlacesAdapter(
 			iconImageView.setImageDrawable(coloredIcon)
 			if (shouldShowImage()) {
 				AndroidUiHelper.updateVisibility(imageView, true)
+				imageView.scaleType = ImageView.ScaleType.CENTER
 				val picasso = PicassoUtils.getPicasso(app)
-				item.wikiImageStubUrl?.let {
-					val creator = Picasso.get()
-						.load(it)
-					if (coloredIcon != null) {
-						creator.error(coloredIcon)
-					}
-					creator.into(imageView, object : Callback {
-						override fun onSuccess() {
-							picasso.setResultLoaded(it, true)
-						}
-
-						override fun onError(e: Exception?) {
-							picasso.setResultLoaded(it, true)
+				if (imageView.tag != item.wikiImageStubUrl) {
+					imageView.tag = item.wikiImageStubUrl
+					item.wikiImageStubUrl?.let {
+						val creator = Picasso.get()
+							.load(it)
+						if (coloredIcon != null) {
+							val errorImageSize =
+								app.resources.getDimension(R.dimen.nearby_place_error_mange_size)
+							val errorDrawable = AndroidUtils.createScaledBitmap(
```

Comment:
It looks complicated/inefficient. Do we always need to scale errorDrawable for every item?